File: python/프로그래머스/두원사이의정수쌍.py
import math
def solution(r1, r2):
    answer = 0
    def cal(target, x):
        return math.sqrt(target**2 - x**2)
    
    for x in range(1, r2 + 1):
        if x < r1:
            temp = cal(r1, x)
            y1 = math.floor(temp)
            # 내부 원 경계상의 점이 정수일 경우
            if temp == y1:
                y1 -= 1
        else:
            y1 = 0
            answer += 1  # x축상의 점 (x,0)을 포함
        
        y2 = math.floor(cal(r2, x))
        answer += (y2 - y1)
    return answer * 4

File: java/두원사이의정수쌍.java
import java.util.*;

class Solution {
    public long solution(int r1, int r2) {
        long answer = 0;
        long rr1=(long)r1*r1;
        long rr2=(long)r2*r2;
        for(int i=1;i<=r2;i++) {
            long d = caly(i, rr1);
            long u = caly(i, rr2);
            if(d == 0 || d*d + (long)i*i == rr1) answer++;
            answer += (u-d);
        }
        
        return answer*4;
    }
    
    long caly(int x, long r) {        
        long xx = (long)x*x;
        if(xx>=r) return 0;
        long yy = r-xx;
        return (long)Math.sqrt(yy);
    }
}
